### ISSUE TITLE
Svelte popup rendering is broken on Firefox

### ISSUE THREAD

**Issue by Crazypersonalph** [BetterSEQTA/BetterSEQTA-Plus#166]
On Firefox, the function `addExtensionSettings`, currently errors out when using it in Firefox.

This is due to how the `renderSvelte` function call inside of that function which then in turn calls the Svelte native function `mount` is erroring out for some unknown reason.

Upon `mount` being called, it throws this error:
```
TypeError: can't access property "constructor", proto is null
```

Stack trace:
```
TypeError: can't access property "constructor", proto is null
    get_setters moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:2199
    set_attributes moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:2103
    default moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:9123
    update_reaction moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:920
    update_effect moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:1011
    create_effect moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:488
    render_effect moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:584
    template_effect moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:587
    default moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:9123
    slot moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:1989
    default moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:9044
    slot moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:1989
    default moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:6733
    slot moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:1989
    default moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:6569
    slot moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:1989
    default moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:6526
    slot moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:1989
    UseInitialMotionValues moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:6467
    UseStyle moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:6510
    UseHTMLProps moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:6553
    UseRender moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:6717
    effect2 moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:2003
    update_reaction moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:920
    update_effect moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:1011
    create_effect moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:488
    branch moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:593
    component moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:2003
    update_reaction moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:920
    update_effect moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:1011
    create_effect moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:488
    block moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:590
    component moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:1996
    UseRender moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:6716
    children moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:9017
    slot moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:1989
    MotionContextProvider moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:6312
    default moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:9010
    slot moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:1989
    UseFeatures moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:6281
    consequent_effect moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:1667
    update_reaction moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:920
    update_effect moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:1011
    create_effect moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:488
    branch moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:593
    if_block moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:1667
    update_reaction moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:920
    update_effect moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:1011
    create_effect moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:488
    block moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:590
    if_block moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:1661
    UseFeatures moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:6278
    default moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:8997
    slot moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:1989
    UseVisualElement moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:6197
    default moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:8982
    slot moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:1989
    UseVisualState moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:7333
    default moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:8969
    slot moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:1989
    UseCreateMotionContext moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:7254
    children moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:8958
    slot moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:1989
    ScaleCorrectionProvider moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:4045
    MotionSSR moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:8955
    MotionDiv moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:9113
    TabbedContainer moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:9173
    Settings_1 moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:30408
    unmount moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:1628
    update_reaction moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:920
    update_effect moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:1011
    create_effect moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:488
    branch moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:593
    unmount moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:1616
    update_reaction moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:920
    update_effect moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:1011
    create_effect moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:488
    effect_root moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:540
    _mount moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:1615
    mount moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:1590
    renderSvelte moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:28285
    addExtensionSettings moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:29184
    AddBetterSEQTAElements moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:25020
    LoadPageElements moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:28828
    tryLoad moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:29027
    promise callback*tryLoad moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:29026
    main moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:29107
    init moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:28327
    async* moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:28304
[settings-Cfzed1-0.js:29186:13](moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js)
    addExtensionSettings moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:29186
    AddBetterSEQTAElements moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:25020
    LoadPageElements moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:28828
    tryLoad moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:29027
    (Async: promise callback)
    tryLoad moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:29026
    main moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:29107
    init moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:28327
    <anonymous> moz-extension://fb504548-cb1e-4053-899d-48cf99628d19/assets/settings-Cfzed1-0.js:28304
```

**Comment by Crazypersonalph** [comment]
React popups, however, do work in Firefox, although with broken animations.

**Comment by SethBurkart123** [comment]
may need to test with rendering a minimal svelte component that doesn't import anything. Try and narrow it down.

**Comment by Crazypersonalph** [comment]
Interestingly enough, on both Chrome and Firefox, the same issue presents when you try to access the popup directly through the extension page.

```
Uncaught TypeError: Cannot read properties of null (reading 'querySelector')
```

**Comment by Crazypersonalph** [comment]
Made a very barebones version that works properly 7acc6e6

**Comment by Crazypersonalph** [comment]
@SethBurkart123 The issue seems to lie in the TabbedContainer component afaik

**Comment by Crazypersonalph** [comment]
I have widdled it down further to it being inside this block of code:

```
<div bind:this={containerRef} class="top-0 z-10 text-[0.875rem] pb-0.5 mx-4 tab-width-container">
    <div class="relative flex">
      <MotionDiv
        class="absolute top-0 left-0 z-0 h-full bg-[#DDDDDD] dark:bg-[#38373D] rounded-full opacity-40 tab-width"
        animate={{ x: calcXPos(hoveredTab) }}
        transition={springTransition}
      />
      {#each tabs as { title }, index}
        <button
          class="relative z-10 flex-1 px-4 py-2"
          onclick={() => activeTab = index}
          onmouseenter={() => hoveredTab = index}
          onmouseleave={() => hoveredTab = null}
        >
          {title}
        </button>
      {/each}
    </div>
  </div>
  <div class="h-full px-4 overflow-y-scroll overflow-x-clip">
    <MotionDiv
      class="h-full"
      animate={{ x: `${-activeTab * 100}%` }}
      transition={springTransition}
    >
      <div class="flex">
        {#each tabs as { Content }, index}
          <div class="absolute w-full transition-opacity duration-300 overflow-y-scroll {activeTab === index ? 'opacity-100' : 'opacity-0'}"
            style="left: {index * 100}%;">
            <Content />
          </div>
        {/each}
      </div>
    </MotionDiv>
  </div>
  ```

**Comment by Crazypersonalph** [comment]
Seems like the issue occurs when the `<Content />` and `<MotionDiv>` tags are used, so it may also be occurring in other places, but we should fix one thing at a time.

**Comment by Crazypersonalph** [comment]
Issue is to do with the native Svelte `get_setters` function

I'm not exactly sure whether this is a problem with our code, or a Svelte 5 bug.

Maybe worth reporting?

```
function get_setters(element) {
  var setters = [];
  var descriptors;
  var proto = get_prototype_of(element);
  while (proto.constructor.name !== "Element") {
    descriptors = get_descriptors(proto);
    for (var key in descriptors) {
      if (descriptors[key].set && !always_set_through_set_attribute.includes(key)) {
        setters.push(key);
      }
    }
    proto = get_prototype_of(proto);
  }
  return setters;
}
```

**Comment by Crazypersonalph** [comment]
For some reason, the prototype of element is null.

**Comment by SethBurkart123** [comment]
Fixed as of [3388281](https://github.com/BetterSEQTA/BetterSEQTA-Plus/commit/33882817445eed1c9bed0ffa6b02d1bf1b27649e)
